pyccel/parser/syntactic.py

```diff
@@ -589,11 +589,13 @@ def _visit_Compare(self, stmt):
                       severity='fatal')
 
     def _visit_Return(self, stmt):
-        results = self._visit(stmt.value)
-        if not isinstance(results, (list, PythonTuple, PythonList)):
-            results = [results]
-        expr = Return(results)
-        return expr
+        if stmt.value is None:
+            return Return([])
+        else:
+            results = self._visit(stmt.value)
+            if not isinstance(results, (list, PythonTuple, PythonList)):
+                results = [results]
+            return Return(results)
```

Comment:
I am not sure of this way to fix issue #900. In Python `return` is totally equivalent to `return None` and so we should be able to handle both cases in exactly the same way.

Reply:
I agree that it is the same in python, but it would not be the same in Fortran/C, I imagine it would be approximately equivalent to `return NULL` with the return type becoming a pointer (but this seems hard/dangerous to implement).

If we consider that `return` and `return None` are strictly equivalent then I could change the condition to `if results is Nil()`